VideoLocker/src/main/java/org/edx/mobile/authentication/LoginAPI.java

```diff
@@ -1,38 +1,173 @@
 
+import org.edx.mobile.exception.AuthException;
+import org.edx.mobile.http.HttpResponseStatusException;
 import org.edx.mobile.http.RetroHttpException;
-import org.edx.mobile.logger.Logger;
+import org.edx.mobile.model.api.RegisterResponse;
+import org.edx.mobile.module.analytics.ISegment;
+import org.edx.mobile.module.notification.NotificationDelegate;
+import org.edx.mobile.module.prefs.LoginPrefs;
+import org.edx.mobile.services.ServiceManager;
 import org.edx.mobile.util.Config;
+import org.edx.mobile.util.observer.BasicObservable;
+import org.edx.mobile.util.observer.Observable;
 
 import retrofit.RestAdapter;
 
-/**
- * Created by cleeedx on 4/12/16.
- */
-
 @Singleton
 public class LoginAPI {
 
     @NonNull
     private final LoginService loginService;
 
-    @Inject
-    Config config;
+    @NonNull
+    private final Config config;
+
+    @NonNull
+    private final LoginPrefs loginPrefs;
+
+    @NonNull
+    private final ServiceManager serviceManager;
+
+    @NonNull
+    private final ISegment segment;
+
+    @NonNull
+    private final NotificationDelegate notificationDelegate;
+
+    @NonNull
+    private final BasicObservable<LogInEvent> logInEvents = new BasicObservable<>();
 
     @Inject
-    public LoginAPI(@NonNull RestAdapter restAdapter) {
+    public LoginAPI(@NonNull RestAdapter restAdapter, @NonNull Config config, @NonNull LoginPrefs loginPrefs, @NonNull ServiceManager serviceManager, @NonNull ISegment segment, @NonNull NotificationDelegate notificationDelegate) {
+        this.config = config;
+        this.loginPrefs = loginPrefs;
+        this.serviceManager = serviceManager;
+        this.segment = segment;
+        this.notificationDelegate = notificationDelegate;
         loginService = restAdapter.create(LoginService.class);
     }
 
+    @NonNull
     public AuthResponse getAccessToken(@NonNull String username,
                                        @NonNull String password) throws RetroHttpException {
         String grantType = "password";
         String clientID = config.getOAuthClientId();
         return loginService.getAccessToken(grantType, clientID, username, password);
     }
+
+    @NonNull
+    public AuthResponse logInUsingEmail(String email, String password) throws Exception {
+        try {
+            final AuthResponse response = getAccessToken(email, password);
+            if (!response.isSuccess()) {
+                throw new AuthException(response.error);
+            }
+            finishLogIn(response, LoginPrefs.AuthBackend.PASSWORD, email.trim());
+            return response;
+        } catch (HttpResponseStatusException ex) {
+            if (ex.getStatusCode() >= 400 && ex.getStatusCode() < 500) {
+                throw new AuthException(ex);
+            }
+            throw ex;
+        }
+    }
+
+    @NonNull
+    public AuthResponse logInUsingFacebook(String accessToken) throws Exception {
+        final AuthResponse response = serviceManager.loginByFacebook(accessToken);
```

Comment:
Hmm, you mean move the calls into `LoginService` and out of ServiceManager? that's a good idea